### PR TITLE
Wildcard userpage

### PR DESCRIPTION
**Pull request by drewbern**
Load userpage using wildcard URL and serverside rendering
Still WIP

### CODE REVIEW

File: src/main/webapp/user-page.html [hunk 1]
```diff
@@ -32,16 +32,6 @@
...
-      <br/>
-      <textarea name="text" placeholder="Enter a message" id="message-input"></textarea>
-      <br/>
-      <input type="submit" value="Submit">
-    </form>
-    <hr/>
-  
     <b>About Me:</b>
     <div id="about-me-container">Loading...</div>
     <br/>
```
Comment (nbaldy) [inline src/main/webapp/user-page.html]: May possibly want to also get rid of this about-me-container, for consistency? (Unless there is a reason for it to still be, in which case that is more important)

**Review by nbaldy — COMMENTED** [review]
This all looks good, I can't completely tell the difference between your additions and current master because it isn't merged though. When you get the chance, can you merge the files locally and update your pull request, so that I can double check against current master easier (and so that we can both make sure that we don't accidentally break anything in the merge?)

File: src/main/webapp/user-page.html (hunk 1 above)
Comment (nbaldy) [reply src/main/webapp/user-page.html]: Also wondering if we still need this file, if we now have a user-page.jsp (legitimately not sure)

Comment (drewbern) [reply src/main/webapp/user-page.html]: Just merged! And yes, thank you for pointing that out! I had forgotten to remove the file. I had a couple of strange merge conflicts, but I think it should be resolved now. Thank you for checking over my PR!

File: src/main/webapp/error.jsp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    </font>
+    <nav>
+      <ul id="navigation">
+        <li><a href="/">Home</a></li>
+    <%
+      if (isUserLoggedIn) {
+        String username = (String) request.getAttribute("username");
+    %>
+        <a href="/user-page.html?user=<%= username %>">Your Page</a>
+    <% } %>
```
Comment (nbaldy) [inline src/main/webapp/error.jsp:24]: user-page.html hre

File: src/main/webapp/index.jsp [hunk 3]
```diff
@@ -18,10 +18,10 @@
       if (isUserLoggedIn) {
         String username = (String) request.getAttribute("username");
     %>
-        <li><a href="/users/<%= username %>">Your Page</a></li>
-        <li><a href="/logout">Logout</a></li>
+        <a href="/user-page.html?user=<%= username %>">Your Page</a>
```
Comment (nbaldy) [inline src/main/webapp/index.jsp]: and here

Comment (drewbern) [reply src/main/webapp/index.jsp]: Fixed!

File: src/main/webapp/error.jsp (hunk 2 above)
Comment (drewbern) [reply src/main/webapp/error.jsp:24]: Fixed!

File: src/main/java/com/google/codeu/servlets/UsersServlet.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    // Fetch user messages
+    List<Message> messages = datastore.getMessages(user);
+    request.setAttribute("isUserLoggedIn", userService.isUserLoggedIn());
+    User userData = datastore.getUser(user);
+    String aboutMe = userData.getAboutMe();
+    if (userData == null || userData.getAboutMe() == null) {
+        aboutMe = "This \"About me\" page is empty :(";
+    }
```
Comment (nbaldy) [inline src/main/java/com/google/codeu/servlets/UsersServlet.java:52]: // Fetch user messages
    List<Message> messages = datastore.getMessages(user);
    request.setAttribute("isUserLoggedIn", userService.isUserLoggedIn());
    User userData = datastore.getUser(user);
    String aboutMe = "This \"About me\" page is empty :(";
    if (userData != null && userData.getAboutMe() != null) {
      aboutMe = userData.getAboutMe();
    }

Comment (drewbern) [reply src/main/java/com/google/codeu/servlets/UsersServlet.java:52]: Fixed!

File: src/main/webapp/js/user-page-loader.js [hunk 5]
```diff
@@ -20,105 +20,7 @@ const parameterUsername = urlParams.get('user');
 
 // URL must include ?user=XYZ parameter. If not, redirect to homepage.
 if (!parameterUsername) {
```
Comment (nbaldy) [inline src/main/webapp/js/user-page-loader.js]: It doesn't look like we use this if statement, if so should we just remove it?

**Review by nbaldy — APPROVED** [review]
Looks great with one possible change, see below

File: src/main/webapp/js/user-page-loader.js (hunk 5 above)
Comment (drewbern) [reply src/main/webapp/js/user-page-loader.js]: Oops forgot that, thanks for spotting it!

**Comment by drewbern** [comment]
#42

**Pull request merged**